packages/core/src/js/sdk.tsx

```diff
@@ -62,6 +63,32 @@ export function init(passedOptions: ReactNativeOptions): void {
     enableSyncToNative(getIsolationScope());
   }
 
+  const userBeforeBreadcrumb = safeFactory(passedOptions.beforeBreadcrumb, { loggerMessage: 'The beforeBreadcrumb threw an error' });
+
+  // Exclude Dev Server and Sentry Dsn request from Breadcrumbs
+  const devServerUrl = getDevServer()?.url;
+  const dsn = passedOptions.dsn;
+  const defaultBeforeBreadcrumb = (breadcrumb: Breadcrumb, _hint?: BreadcrumbHint): Breadcrumb | null => {
+    const type = breadcrumb.type || '';
+    const url = typeof breadcrumb.data?.url === 'string' ? breadcrumb.data.url : '';
+    if (type === 'http' && ((devServerUrl && url.includes(devServerUrl)) || (dsn && url.includes(dsn)))) {
```

Comment:
q: is the DSN filter going to work?
Usually, the DSN is not exactly on [URL format](https://docs.sentry.io/concepts/key-terms/dsn-explainer/#the-parts-of-the-dsn), 

If you look at the picture [here](https://github.com/getsentry/sentry-react-native/issues/3105) the final format is different.

It may be more complex, but one way we could do this is to use some similar code to extract the url like here:
https://github.com/getsentry/sentry-javascript/blob/f4c59000ca57b566c3ed05fe5a0b06acd75a2543/packages/core/src/baseclient.ts#L136-L146

Reply:
Good point @lucas-zimerman 👍 
I've updated the code with a69d5e8cd2f78882e237bba3e15fdadd090d6fab to extract the url from the passed dsn string for filtering the breadcrumbs.
Thank you for catching this and all the help 🙇